Improve `RecentEra` docs

Anviking opened the pull request:
Related Advice Process Decision: https://docs.google.com/document/d/1xazQNECWex3GWp2-yzAeJGnYvAHchamgNVvFDs3ydcg/edit?tab=t.0#heading=h.gjdgxs

Anviking committed:
Improve `RecentEra` docs

HeinrichApfelmus commented on lib/balance-tx/lib/internal/Internal/Cardano/Write/Eras.hs:
```diff
@@ -110,13 +110,37 @@ type LatestLedgerEra = Conway
+-- myTx :: IsRecentEra era => Tx era
+-- signByAlice :: IsRecentEra era => Tx era -> Tx era
+-- renderCliTx :: IsRecentEra era -> Tx era -> Text
+-- myCardanoApiTx :: IsRecentEra era -> RecentEra -> CardanoApi.Tx (CardanoApiEra era)
```
```suggestion
-- renderCliTx :: IsRecentEra era => Tx era -> Text
-- myCardanoApiTx :: IsRecentEra era => RecentEra era → CardanoApi.Tx (CardanoApiEra era)
```

The other variants do not type check, I believe.

Anviking committed:
Copy `RecentEra` vs `IsRecentEra` guidelines verbatim

From advice process decision [Singleton type RecentEra :: AP](https://docs.google.com/document/d/1xazQNECWex3GWp2-yzAeJGnYvAHchamgNVvFDs3ydcg/edit?tab=t.0#heading=h.gjdgxs)

Co-authored-by: Heinrich Apfelmus <apfelmus@quantentunnel.de>

HeinrichApfelmus reviewed (approved):
Thanks! 😊 👍
